frontend/app/controllers/TierController.scala

```diff
@@ -43,19 +42,30 @@ trait UpgradeTier {
 
   def upgrade(tier: Tier) = MemberAction.async { implicit request =>
 
-    def futureCustomerOpt = request.member match {
+    def futurePaidPreviewOpt = request.member match {
       case paidMember: PaidMember =>
-        request.touchpointBackend.stripeService.Customer.read(paidMember.stripeCustomerId).map(Some(_))
+        for {
+          preview <- MemberService.previewUpgradeSubscription(paidMember, request.user, tier)
+          customer <- request.touchpointBackend.stripeService.Customer.read(paidMember.stripeCustomerId)
```

Comment:
oooh interesting. I'll change this to instantiate futures before the for comprehension.